lib/array_equals.rb

```diff
@@ -1,5 +1,18 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil && array2 == nil
+    return true
+  elsif array1 == nil || array2 == nil || array1.count != array2.count
+    return false
+  else
```

Comment:
You do not need this `else`. If you have not returned in line 4 or line 7, you will hit this line.